### PR TITLE
Updating the issue template to add validation checks

### PR DESCRIPTION
**Pull request by bbarani**
### Description
Updating the issue templates to add validation
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/480 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_template.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+labels: [bug]
+body:
+ - type: markdown
+   attributes:
+    value: |
+     Thanks for taking the time to fill out this bug report!
+ - type: textarea
+   id: what-happened
+   attributes:
+    label: What happened?
```
Comment (gaiksaya) [inline .github/ISSUE_TEMPLATE/bug_template.yaml:13]: nit: Can we say `Describe the bug` instead of `What happened`?

**Comment by abhinavGupta16** [comment]
I believe we should also have Acceptance Criteria as a Mandatory field as a part of the template.

```
Acceptance Criteria -

Given ...
When ....
Then ....
```

We could use this or any other template which suits our needs better.

File: .github/ISSUE_TEMPLATE/bug_template.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    description: Also tell us, what did you expect to happen?
```
Comment (gaiksaya) [inline .github/ISSUE_TEMPLATE/bug_template.yaml:14]: We can have something like `Acceptance Criteria` or `Expected behavior` as one more mandatory. Helps us to get a clean requirement and expectation of the customer

File: .github/ISSUE_TEMPLATE/bug_template.yaml (hunk 1 above)
Comment (peterzhuamazon) [reply .github/ISSUE_TEMPLATE/bug_template.yaml:13]: +1

File: .github/ISSUE_TEMPLATE/bug_template.yaml (hunk 2 above)
Comment (peterzhuamazon) [reply .github/ISSUE_TEMPLATE/bug_template.yaml:14]: +1

File: .github/ISSUE_TEMPLATE/bug_template.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: Bug Report
+description: File a bug report
+title: "[Bug]: "
+labels: [bug]
```
Comment (peterzhuamazon) [inline .github/ISSUE_TEMPLATE/bug_template.yaml:4]: Would be better to have untriaged label back: `labels: 'bug, untriaged'`

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: Feature request
+description: Suggest an idea for this project
+labels: [enhancement]
```
Comment (peterzhuamazon) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:3]: labels: 'enhancement, untriaged'

**Comment by dblock** [comment]
This deviates from what we have in .github, are we concerned about 1) de-standardizing 2) making it more difficult for users to open issues?

**Comment by bbarani** [comment]
@dblock I have made the template almost same as the one we have in .github but added mandatory validation for 2 fields. All other fields are optional.

File: .github/ISSUE_TEMPLATE/bug_template.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+     Thanks for taking the time to fill out this bug report!
+ - type: textarea
+   id: what-happened
+   attributes:
+    label: Describe the bug
+    description: A clear and concise description of what the bug is.
+   validations:
+    required: true
+ - type: textarea
+   id: to-Reproduce
```
Comment (gaiksaya) [inline .github/ISSUE_TEMPLATE/bug_template.yaml:18]: I think you missed the `Steps to reproduce` section.

**Review by peterzhuamazon — APPROVED** [review]

**Review by peternied — APPROVED** [review]
Great improve in our bug templates - can we see about rolling this out to the organization defaults?

**Pull request merged**